packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -156,8 +171,56 @@ const clampTimeRange = (timeRange: ITimeRange): ITimeRange => {
   };
 };
 
-const generateDefaultTime = (increments: number, timeRange: ITimeRange | undefined) => {
-  const newDefaultTime = new Date();
+const getDateFromSelection = (showSeconds: boolean, useHour12: boolean, defaultTime: Date, selectedTime: string) => {
```

Comment:
I feel like this function could be moved out of the component?

Reply:
Resolved by moving function out of component.